test/e2e/grpc/client.test.ts

```diff
@@ -178,6 +184,34 @@ describe('grpc/client', () => {
       expect(res).toEqual('value-1');
     });
 
+    it('should be able to add metadata, etag and options', async () => {
+      await client.state.save('state-redis', [
+        {
+          key: 'key-1',
+          value: 'value-1',
+          etag: "1234",
+          options: {
+            "concurrency": "first-write",
+            "consistency": "strong"
+          },
+          metadata: {
+            hello: "world"
+          }
+        },
+        {
+          key: 'key-2',
+          value: 'value-2',
+        },
+        {
+          key: 'key-3',
+          value: 'value-3',
+        },
+      ]);
+
+      const res = await client.state.get('state-redis', 'key-1');
```

Comment:
In a scenario where we discard the etag, options and metadata, this test will still continue to work, right? Can we change that? Or if that requires some other work or is out of scope for this PR, we can track that separately.

Reply:
There are other tests not using these, so it should normally?